### PR TITLE
Update rubocop

### PR DESCRIPTION
**Pull request by groyoh**
#### Purpose
Rebase #1498 

#### Related GitHub issues
Closes #1498 

#### Additional helpful information

@bf4 @remear here you go :smile:  Just need to update the trailling/leading dot and add a CHANGELOG. Anything else missing?

### CODE REVIEW

File: lib/active_model_serializers/adapter.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module ActiveModelSerializers
   module Adapter
     UnknownAdapterError = Class.new(ArgumentError)
-    ADAPTER_MAP = {}
+    ADAPTER_MAP = {} # rubocop:disable Style/MutableConstant
```
Comment (groyoh) [inline lib/active_model_serializers/adapter.rb]: Rubocop changed this to:
```ruby
ADAPTER_MAP = {}.freeze
```

which prevent us to modify the hash at [line 52](https://github.com/groyoh/active_model_serializers/blob/update_rubocop/lib/active_model_serializers/adapter.rb#L52) and thus breaks the gem/tests.

Comment (bf4) [inline lib/active_model_serializers/adapter.rb]: :)

**Comment by bf4** [comment]
```
test/action_controller/serialization_test.rb:435:7: W: Lint/UnneededDisable: Unnecessary disabling of Lint/DuplicateMethods.
      # rubocop:disable Lint/DuplicateMethods
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
148 files inspected, 1 offense detected
```

womp wmp

**Comment by groyoh** [comment]
Weird, rubocop is green locally :confused:

**Comment by groyoh** [comment]
Ah found it, it was `0.37` locally but `0.37.2` on Travis.

**Comment by groyoh** [comment]
@bf4 What do you think about pinning the rubocop version?

**Comment by bf4** [comment]
Depends if it's broken and causing trouble or just different 

B mobile phone

**Comment by groyoh** [comment]
I assume new cops are added every few versions so if the version is not fixed, then we might run into new cops every now and then, making the build fail.

File: .rubocop.yml [hunk 2]
```diff
@@ -59,6 +64,18 @@ Style/BlockDelimiters:
...
+  EnforcedStyle: semantic
+
+Style/TrailingCommaInLiteral:
+  EnforcedStyleForMultiline: consistent_comma
+
+Style/ConditionalAssignment:
+  Enabled: false
+
+Style/DotPosition:
+  EnforcedStyle: trailing
```
Comment (groyoh) [inline .rubocop.yml]: As @remear I tend to prefer leading here. what about you @bf4?

**Comment by NullVoxPopuli** [comment]
needs rebase -- as long as the checks pass, I'm good with merging.

File: .rubocop.yml [hunk 3]
```diff
@@ -59,6 +64,18 @@ Style/BlockDelimiters:
   Enabled: true
   EnforcedStyle: line_count_based
 
+Style/SignalException:
+  EnforcedStyle: semantic
+
+Style/TrailingCommaInLiteral:
+  EnforcedStyleForMultiline: consistent_comma
```
Comment (remear) [inline .rubocop.yml]: Is the trailing comma what all the cool kids do these days? https://github.com/rails-api/active_model_serializers/pull/1560/files#diff-57dc91570125bb0a036a83a5c8bd6be8R63

Usually I'm fine with it being allowed, not necessarily required, as is a habit I picked up from Python for this. I'm curious if others have any strong opinion here.

Comment (NullVoxPopuli) [inline .rubocop.yml]: I'm personally not a fan of the trailing comma. But I guess if that's what the standard is now a days...

**Comment by remear** [comment]
Sounds like the team prefers the `DotPosition` before, e.g.,

```ruby
SomeClass.awesome_sauce
 .do_things
```

The team also prefers to not enforce a trailing comma.

**Comment by groyoh** [comment]
Rebased and updated with the style fix.

**Comment by remear** [comment]
:+1:

**Comment by NullVoxPopuli** [comment]
I'm good with this.

**Comment by NullVoxPopuli** [comment]
do we want to merge anything before we merge this, as it could cause conflicts in other PRs?

**Comment by groyoh** [comment]
@NullVoxPopuli good point here. It would be easier to merge this first and update the other PR as this one is pretty big.

**Pull request merged**